griptape/artifacts/list_artifact.py

```diff
@@ -31,18 +34,21 @@ def child_type(self) -> Optional[type]:
         else:
             return None
 
-    def __getitem__(self, key: int) -> BaseArtifact:
+    def __getitem__(self, key: int) -> T:
         return self.value[key]
 
     def __bool__(self) -> bool:
         return len(self) > 0
 
+    def __add__(self, other: BaseArtifact) -> ListArtifact[T]:
```

Comment:
should this be `T` as well?

Reply:
No, otherwise the override will be incompatible with the base class.

Reply:
feels inconsistent then

Reply:
A generic does not belong here. When adding something to the `ListArtifact`, it should be a `BaseArtifact`, not whatever we define in the generic.

From pyright:
```
Method "__add__" overrides class "BaseArtifact" in an incompatible manner
     Parameter 2 type mismatch: base parameter is type "BaseArtifact", override parameter is type "T@ListArtifact"
       Type "BaseArtifact" is incompatible with type "T@ListArtifact" [reportIncompatibleMethodOverride]
```